test/e2e/cluster/add_remove_cluster.go

```diff
@@ -0,0 +1,53 @@
+package cluster
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/submariner-io/submariner/test/e2e/dataplane"
+	"github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var _ = Describe("[expansion] Test expanding/shrinking an existing cluster fleet", func() {
+	f := framework.NewDefaultFramework("add-remove-cluster")
+
+	It("Should be able to add and remove third cluster", func() {
+		clusterAName := framework.TestContext.KubeContexts[framework.ClusterA]
+		clusterBName := framework.TestContext.KubeContexts[framework.ClusterB]
+		clusterCName := framework.TestContext.KubeContexts[framework.ClusterC]
+
+		By(fmt.Sprintf("Verifying no GW nodes are present on cluster %q", clusterCName))
+		gatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, true)
+		Expect(gatewayNode).To(HaveLen(0), fmt.Sprintf("Expected no gateway node on %q", framework.ClusterC))
+
+		By(fmt.Sprintf("Verifying that a pod in cluster %q cannot connect to a pod in cluster %q", clusterCName, clusterAName))
+		dataplane.RunNoConnectivityTest(f, false, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterC, framework.ClusterA)
+
+		By(fmt.Sprintf("Verifying that a pod in cluster %q cannot connect to a service in cluster %q", clusterCName, clusterBName))
+		dataplane.RunNoConnectivityTest(f, true, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterC, framework.ClusterB)
+
+		nonGatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, false)
+		Expect(nonGatewayNode).To(HaveLen(2), fmt.Sprintf("Found non-gateway node on %q", clusterCName))
```

Comment:
```suggestion
		Expect(nonGatewayNode).ToNot(HaveLen(0), fmt.Sprintf("No non-gateway nodes found on %q", clusterCName))
```

We really don't need to check for an exact count, only that there's at least one non-GW node.

Reply:
Sprintf message should say that non-GW node is found. So the current implementation stands. I changed the expect condition.

Reply:
Sprintf message should say there are non-GW nodes found. Hence keeping it as it is, Changing expect condition

Reply:
The message is only outputted if the assertion fails which means there aren't any non-GW nodes so saying "Found non-gateway node on ..." is incorrect.

Reply:
Done